test/ctia/stores/es/init_test.clj

```diff
@@ -175,6 +175,23 @@
                    false
                    nil))))))
 
+(deftest ambiguous-indices-test
+  (let [test-cases [{:message "partial and restored should not match the ambiguous index names"
+                     :existing {:v2.0.0_ctia_sighting-2022.07.07-000001 {}
+                                :partial-v2.0.0_ctia_sighting-2022.07.07-000001 {}
+                                :restored-v2.0.0_ctia_sighting-2022.07.07-000001 {}}
+                     :index "v2.0.0_ctia_sighting"
+                     :expected #{}}
+                    {:message "ambiguous names with unknown prefix should be detected"
+                     :existing {:v2.0.0_ctia_sighting-2022.07.07-000001 {}
+                                :prefix-v2.0.0_ctia_sighting-2022.07.07-000001 {}}
+                     :index "v2.0.0_ctia_sighting"
+                     :expected #{:prefix-v2.0.0_ctia_sighting-2022.07.07-000001}}]]
```

Comment:
Can you add a test case that would fail if you don't regex-quote the index?

`.` is probably a regex character, maybe try this:

```clojure
                    {:message "ambiguous names with unknown prefix should be detected"
                     :existing {:v2x0x0_ctia_sighting-2022.07.07-000001 {}
                                :prefix-v2x0x0_ctia_sighting-2022.07.07-000001 {}}
                     :index "v2.0.0_ctia_sighting"
                     :expected #{:prefix-v2.0.0_ctia_sighting-2022.07.07-000001}}
```